src/plugins/data/public/query/query_string/query_string_manager.ts

```diff
@@ -336,7 +336,7 @@ export class QueryStringManager {
 
     return {
       ...newQuery,
-      query: this.getInitialDatasetQueryString(newQuery),
+      query: useInitialDatasetQueryString ? this.getInitialDatasetQueryString(newQuery) : '',
```

Comment:
this is a change in `data` plugin. i'm wondering now that i have submitted this PR whether I shouldn't make this change, and just make the change in the explore plugin... I'll think more about it tomorrow